core/src/main/java/com/linecorp/armeria/server/annotation/decorator/RequestTimeout.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.annotation.decorator;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.concurrent.TimeUnit;
+
+import com.linecorp.armeria.server.annotation.DecoratorFactory;
+
+/**
+ * Annotation for request timeout
+ */
+@DecoratorFactory(RequestTimeoutDecoratorFunction.class)
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface RequestTimeout {
```

Comment:
should this be named RequestTimeoutDecorator, as other decorating annotations are named?

Reply:
From the users' point of view, it is a timeout scheduler. We don't need to tell the implementation detail.